core/vibes/soul/sections/product-detail/product-detail-form.tsx

```diff
@@ -197,16 +200,19 @@ function FormField({
 }) {
   const controls = useInputControl(formField);
 
-  const [, setParams] = useQueryStates(
+  const [params, setParams] = useQueryStates(
     field.persist === true ? { [field.name]: parseAsString.withOptions({ shallow: false }) } : {},
   );
 
   const handleChange = useCallback(
     (value: string) => {
-      void setParams({ [field.name]: value });
-      controls.change(value);
+      // Ensure that if page is reached without a full reload, we are still setting the selection properly based on query params.
+      const fieldValue = value || String(params[field.name]);
+
+      void setParams({ [field.name]: fieldValue });
+      controls.change(fieldValue);
```

Comment:
Is this to fix the issue with the reload?

Reply:
@jorgemoya This is separate - this is a fix for login redirect back to the product page. If you attempt to add a product to a wishlist after selecting options while logged out, you'll go to `/login?redirectTo=/my-product?123=1234`, and then after logging in, you'll be redirected to the product page with the query params. This change here ensures that after the redirect, the product options are re-selected properly.

Reply:
Nice!